### PR TITLE
Propagete node role through machine spec label

### PR DESCRIPTION
**Pull request by ingvagabund**
SSIA

### CODE REVIEW

File: cmd/nodelink-controller/main.go [hunk 1]
```diff
@@ -401,22 +400,6 @@ func (c *Controller) processNode(node *corev1.Node) error {
 		modNode.Labels = map[string]string{}
```
[the remark this answers is not in the document]
Comment (ingvagabund) [reply cmd/nodelink-controller/main.go]: We do propagate labels. Or why do you think otherwise?

**Comment by ingvagabund** [comment]
> Not sure whether we need to create the empty labels slice.

nil is nil after serialization

File: cmd/nodelink-controller/main.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (frobware) [reply cmd/nodelink-controller/main.go]: I missed where we copy them later in the loop.

/lgtm

**Pull request merged**